artifactory/commands/terraform/terraformpublish.go

```diff
@@ -0,0 +1,304 @@
+	commandName string
+	result      *commandsutils.Result
+}
+
+func NewTerraformPublishCommand() *TerraformPublishCommand {
+	return &TerraformPublishCommand{TerraformPublishCommandArgs: NewTerraformPublishCommandArgs(), commandName: "rt_terraform_publish", result: new(commandsutils.Result)}
+}
+
+func NewTerraformPublishCommandArgs() *TerraformPublishCommandArgs {
+	return &TerraformPublishCommandArgs{}
+}
+
+func (npc *TerraformPublishCommand) ServerDetails() (*config.ServerDetails, error) {
+	return npc.serverDetails, nil
+}
+
+func (tpc *TerraformPublishCommand) CommandName() string {
+	return tpc.commandName
+}
+
+func (tpc *TerraformPublishCommand) SetConfigFilePath(configFilePath string) *TerraformPublishCommand {
+	tpc.configFilePath = configFilePath
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) Result() *commandutils.Result {
+	return tpc.result
+}
+
+func (tpc *TerraformPublishCommand) SetNamespace(namespace string) *TerraformPublishCommand {
+	tpc.namespace = namespace
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetModuleName(moduleName string) *TerraformPublishCommand {
+	tpc.moduleName = moduleName
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetProvider(provider string) *TerraformPublishCommand {
+	tpc.provider = provider
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetTag(tag string) *TerraformPublishCommand {
+	tpc.tag = tag
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetExclusions(exclusions []string) *TerraformPublishCommand {
+	tpc.exclusions = exclusions
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) Run() error {
+	log.Info("Running Terraform Publish")
+	err := tpc.preparePrerequisites()
+	if err != nil {
+		return err
+	}
+	err = tpc.publish()
+	if err != nil {
+		return err
+	}
+	log.Info("Terraform publish finished successfully.")
+	return nil
+}
+
+func (tpc *TerraformPublishCommand) preparePrerequisites() error {
+	namespace, provider, tag, exclusions, err := ExtractTerraformPublishOptionsFromArgs(tpc.args)
+	if err != nil {
+		return err
+	}
+	if namespace == "" || provider == "" || tag == "" {
+		return errorutils.CheckErrorf("Wrong number of arguments. for a terraform publish command please provide --namespace, --provider and --tag.")
+	}
+	tpc.SetNamespace(namespace).SetProvider(provider).SetTag(tag).SetExclusions(exclusions)
+	if err := tpc.setRepoFromConfiguration(); err != nil {
+		return err
+	}
+	artDetails, err := tpc.serverDetails.CreateArtAuthConfig()
+	if err != nil {
+		return err
+	}
+	return utils.ValidateRepoExists(tpc.repo, artDetails)
+}
+
+func (tpc *TerraformPublishCommand) publish() error {
+	log.Debug("Deploying terraform module.")
+	success, failed, err := tpc.TerraformPublish()
+	if err != nil {
+		return err
+	}
+	tpc.result.SetSuccessCount(success)
+	tpc.result.SetFailCount(failed)
+	return nil
+}
+
+func ExtractTerraformPublishOptionsFromArgs(args []string) (namespace, provider, tag string, exclusions []string, err error) {
```

Comment:
Looks like this function can be made private. 
In addition, if you made this function a member of the struct, you wouldn't need to pass any arguments to it.
Also, I suggest adding a unit test for this function.